types/koa/test/default.ts

```diff
@@ -33,7 +33,7 @@ app.use<{ a: boolean }>(async (ctx, next) => {
     await next();
 });
 
-app.use((ctx, next) => {
+app.use((ctx: Koa.Context, next) => {
```

Comment:
This change is not related to the subject of this PR and should be done in a separate PR.

Reply:
I understand why you think so, but it really is.

Due to a bug in TypeScript the type of an assertion function or object containing an assertion function must be explicitly types. It doesn’t work if the type is inferred from `Koa.Middleware` or `new Koa().use()`.

```ts
app.use((ctx) => {
  // The result is used, so this is fine
  const a = ctx.throw();

  // This is not an assertion, so this is fine
  ctx.toString();

  // This only works if ctx is explicitly typed
  // Assertions require every name in the call target to be declared with an explicit type annotation. ts(2775)
  ctx.throw();
});
```

Related bugs: [microsoft/TypeScript#34596](https://github.com/microsoft/TypeScript/issues/34596) [microsoft/TypeScript#36931](https://github.com/microsoft/TypeScript/issues/36931)

This is why I initially gave up on #52449

Reply:
Got it! Thanks for in-depth explanation...